annif/vocab.py

```diff
@@ -25,6 +25,24 @@ def _create_subject_index(self, subject_corpus):
         self._subjects = annif.corpus.SubjectIndex(subject_corpus)
         annif.util.atomic_save(self._subjects, self.datadir, 'subjects')
 
+    def _update_subject_index(self, subject_corpus):
+        old_subjects = self.subjects
+        new_subjects = annif.corpus.SubjectIndex(subject_corpus)
+        updated_subjects = annif.corpus.SubjectIndex.load(os.path.devnull)
```

Comment:
It's maybe a bit stupid to create a new variable for updated subjects (at least this way). Another way could be to add `__setitem__` in `SubjectIndex` class, which could be used to update the labels in `old_subjects`. Somehow this now implemented way still seemed simpler to me. What would be better?

Reply:
I think this method of generating the new vocabulary is fine.

I think it's a bit awkward to have to use `os.path.devnull` here. How about making the subject_corpus parameter optional in the SubjectIndex constructor? Then the above line would become just:
    updated_subjects = annif.corpus.SubjectIndex()